src/signature_accumulator.rs

```diff
@@ -212,25 +205,23 @@ mod tests {
             for signature_msg in msg_and_sigs.signature_msgs {
                 let old_num_msgs = sig_accumulator.msgs.len();
 
-                let result = match signature_msg.variant {
-                    Variant::MessageSignature(msg) => sig_accumulator.add_proof(*msg),
+                let result = match signature_msg.variant() {
+                    Variant::MessageSignature(msg) => sig_accumulator.add_proof(*msg.clone()),
```

Comment:
`*` shall not be required anymore?

Reply:
Ah it was because this field is a box, so a deref call to the msg then clone(). I tried without.

Reply:
ah, right.